File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cpp
#include <string>

using namespace std;

int solution(string myString, string pat) {
    int answer = 0;
    
    size_t idx = myString.find(pat);
    while (idx != string::npos) {
        answer++;
        myString = myString.substr(idx + 1);
        idx = myString.find(pat);
    }
    
    return answer;
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.java
import java.util.*;

class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        
        int idx = myString.indexOf(pat);
        
        while (idx != -1) {
            answer++;
            myString = myString.substring(idx + 1);
            idx = myString.indexOf(pat);
        }
        
        return answer;
    }
}